### PR TITLE
Support Path Params in Rewrite Paths with API Policy

### PR DESCRIPTION
**Pull request by renuka-fernando**
### Purpose
This will support a rewrite of

```
/shops/{shopId}/pets/{petId}/orders
to
/pets/{uri.var.petId}/hello-shops/abcd-shops/{uri.var.shopId}
```

### Issues

Fixes https://github.com/wso2/product-microgateway/issues/2745

### Automation tests
 - Unit tests added: Yes
 - Integration tests added: Yes

### Tested environments

Tested in docker compose setup with apictl

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

**Review by pubudu538 — APPROVED** [review]

**Comment by VirajSalaka** [comment]
Can we transform something like `/pet/{petId}.api` ? 
If we do not, simply mentioning it in a doc would be enough I guess.

**Comment by renuka-fernando** [comment]
> Can we transform something like `/pet/{petId}.api` ? to pet/petApis/{petId} If we do not, simply mentioning it in a doc would be enough I guess. If it is supported let's add a unit test there too.

Added unit + integration tests for this URL format with https://github.com/wso2/product-microgateway/pull/2786/commits/d568961b5d4d84ba8de5bb7c583cc93396a44dc3

File: adapter/internal/oasparser/model/common.go [hunk 1]
```diff
@@ -197,3 +198,43 @@ func unmarshalSwaggerResources(path string, methods []*Operation, vendorExtensio
...
 }
+
+// getRewriteRegexFromPathTemplate returns a regex with capture groups for given rewritePathTemplate
+func getRewriteRegexFromPathTemplate(pathTemplate, rewritePathTemplate string) (string, error) {
+	rewriteRegex := "/" + strings.TrimSuffix(strings.TrimPrefix(rewritePathTemplate, "/"), "/")
+	pathParamToIndexMap := getPathParamToIndexMap(pathTemplate)
+	r := regexp.MustCompile(`{uri.var.([^{}]+)}`)
+	matches := r.FindAllStringSubmatch(rewritePathTemplate, -1)
+	for _, match := range matches {
+		if len(match) > 1 {
```
Comment (renuka-fernando) [inline adapter/internal/oasparser/model/common.go:209]: I think we can remove this check and the check in L235, since the length of slice is always two

Comment (renuka-fernando) [reply adapter/internal/oasparser/model/common.go]: Removed with https://github.com/wso2/product-microgateway/pull/2786/commits/86e265b2f8e11e9df24abef4cccd005c1934c386

**Review by VirajSalaka — APPROVED** [review]

**Pull request merged**